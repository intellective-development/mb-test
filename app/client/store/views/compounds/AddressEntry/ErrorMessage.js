// @flow

import React from 'react';
import cn from 'classnames';
import i18n from 'store/localization';

type EntryMessageProps = {|message_type: 'no_address' | 'no_street' | 'not_found' | 'non_suggested_address'|};
const EntryMessage = ({message_type}: EntryMessageProps) => {
  const message = i18n.t(`ui.error.address_entry.${message_type}`, {defaults: [{scope: 'ui.error.address_entry.default'}]});

  return (
    <div className="cm-ae-error">
      <p className={cn('cm-ae-error__text', {'cm-ae-error__text--empty': !message_type})}>
        {message}
      </p>
    </div>
  );
};

export default EntryMessage;
