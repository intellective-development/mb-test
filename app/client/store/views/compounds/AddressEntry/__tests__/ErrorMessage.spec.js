import * as React from 'react';

import TestProvider from 'store/views/__tests__/utils/TestProvider';
import ErrorMessage from '../ErrorMessage';

describe('ErrorMessage', () => {
  it('renders', () => {
    expect(render(
      <TestProvider>
        <ErrorMessage message_type="no_address" />
      </TestProvider>
    )).toMatchSnapshot();
  });
});
