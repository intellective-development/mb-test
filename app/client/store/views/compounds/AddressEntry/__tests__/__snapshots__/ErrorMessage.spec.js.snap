// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders 1`] = `
<div
  class="cm-ae-error"
>
  <p
    class="cm-ae-error__text"
  >
    Please enter a delivery address to shop.
  </p>
</div>
`;
